spt/strafestuff.cpp

```diff
@@ -256,38 +256,99 @@ double YawStrafeMaxAngle(PlayerData& player, const MovementVars& vars, bool ongr
+		out.SideSpeed += vars.Maxspeed;
+	}
+	if (out.Left) {
+		out.SideSpeed -= vars.Maxspeed;
+	}
+}
+
+bool StrafeJump(PlayerData& player, const MovementVars& vars, bool ducking, ProcessedFrame& out)
+{
+	if (tas_strafe_jumptype.GetInt() == 2) {
+		// OE bhop
+		out.Yaw = NormalizeDeg(tas_strafe_yaw.GetFloat());
+		out.Forward = false;
+		out.Back = false;
+		out.Right = false;
+		out.Left = false;
+		return true;
+	}
+	else if (player.Velocity.Length2D() >= vars.Maxspeed * ((ducking || (vars.Maxspeed == 320)) ? 0.1 : 0.5)) {
+		if (tas_strafe_jumptype.GetInt() == 1) {
+			// ABH
+			out.Yaw = NormalizeDeg(tas_strafe_yaw.GetFloat() + 180);
 			out.Forward = false;
 			out.Back = false;
 			out.Right = false;
 			out.Left = false;
-			return onground;
-		} else if (player.Velocity.Length2D() >= vars.Maxspeed * ((ducking || (vars.Maxspeed == 320)) ? 0.1 : 0.5)) {
-			if (tas_strafe_jumptype.GetInt() == 1) {
-				// ABH
-				out.Yaw = NormalizeDeg(tas_strafe_yaw.GetFloat() + 180);
-				out.Forward = false;
-				out.Back = false;
-				out.Right = false;
-				out.Left = false;
-				return onground;
-			} else if (tas_strafe_jumptype.GetInt() == 3) {
-				// Glitchless bhop
-				const Vector vel = player.Velocity;
-				out.Yaw = NormalizeRad(atan2(vel.y, vel.x)) * M_RAD2DEG;
-				out.Forward = false;
-				out.Back = false;
-				out.Right = false;
-				out.Left = false;
-				return onground;
-			}
+			return true;
 		}
+		else if (tas_strafe_jumptype.GetInt() == 3) {
+			// Glitchless bhop
+			const Vector vel = player.Velocity;
+			out.Yaw = NormalizeRad(atan2(vel.y, vel.x)) * M_RAD2DEG;
+			out.Forward = false;
+			out.Back = false;
+			out.Right = false;
+			out.Left = false;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+
+void StrafeVectorial(PlayerData& player, const MovementVars& vars, bool onground, bool jumped, bool ducking, StrafeType type, StrafeDir dir, double target_yaw, double vel_yaw, ProcessedFrame& out, bool reduceWishspeed)
+{
+	if (jumped && StrafeJump(player, vars, ducking, out)) {
+		MapSpeeds(out, vars);
+		return;
+	}
+
+	ProcessedFrame dummy;
+	Strafe(player, vars, onground, jumped, ducking, type, dir, target_yaw, vel_yaw, dummy, reduceWishspeed, StrafeButtons(), true); // Get the desired strafe direction by calling the Strafe function while using forward strafe buttons
+
+	// If forward is pressed, strafing should occur
+	if (dummy.Forward)
+	{
+		// Calculate updated yaw
+		double normalizedDiff = NormalizeDeg(target_yaw - vel_yaw + tas_strafe_vectorial_offset.GetFloat());
+		double additionAbs = std::min(static_cast<double>(tas_strafe_vectorial_increment.GetFloat()), std::abs(normalizedDiff));
+		double additionSign = (normalizedDiff > 0) ? 1 : -1;
+
+		// Snap to target if difference too large(likely due to an ABH)
+		if (std::abs(normalizedDiff) > tas_strafe_vectorial_snap.GetFloat() && tas_strafe_vectorial_increment.GetFloat() > 0)
+			out.Yaw = target_yaw;
+		else
+			out.Yaw = vel_yaw + additionAbs * additionSign;
```

Comment:
There's a hot [copysign](https://en.cppreference.com/w/cpp/numeric/math/copysign) function for this although I don't think it matters much.